internal/pkg/agent/cmd/inspect.go

```diff
@@ -135,13 +141,47 @@ func inspectConfig(ctx context.Context, cfgPath string, opts inspectConfigOpts,
 		return err
 	}
 
-	if !opts.variables {
+	if !opts.variables && !opts.includeMonitoring {
 		return printConfig(fullCfg, l, streams)
 	}
 	cfg, err := getConfigWithVariables(ctx, l, cfgPath, opts.variablesWait)
 	if err != nil {
 		return err
 	}
+
+	if opts.includeMonitoring {
+		// Load the requirements before trying to load the configuration. These should always load
+		// even if the configuration is wrong.
+		platform, err := component.LoadPlatformDetail()
+		if err != nil {
+			return fmt.Errorf("failed to gather system information: %w", err)
+		}
+		specs, err := component.LoadRuntimeSpecs(paths.Components(), platform)
+		if err != nil {
+			return fmt.Errorf("failed to detect inputs and outputs: %w", err)
+		}
+
+		monitorFn, err := getMonitoringFn(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to get monitoring: %w", err)
+		}
+		_, binaryMapping, err := specs.PolicyToComponents(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to get binary mappings: %w", err)
+		}
+		monitorCfg, err := monitorFn(cfg, binaryMapping)
+		if err != nil {
+			return fmt.Errorf("failed to get monitoring config: %w", err)
+		}
+
+		if monitorCfg != nil {
+			// print separate rather than merge.
+			if err := printMapStringConfig(monitorCfg, streams); err != nil {
```

Comment:
Why seperate instead of the merge? Will that be clear in the output of the command?

Reply:
will merge